Merge 10.x-dev to 10.x for new release

neilblair opened the pull request:
Hoping to get Adam's fixes for Origins Translations menu styling released.  Not sure if other changes here are safe to release?

The latest tag is 10.5.0.  The latest release version is 10.4.9.  Currently nidirect is using the 10.5.0 tag.  So I assume the next release version should be 10.5.1?

omahm committed:
DEPT-1172 ~ Hilight individual fields with duplicate value

omahm committed:
Merge pull request #397 from dof-dss/DEPT-1172

DEPT-1172 ~ Hilight individual fields with duplicate value

soda736 committed:
Check user has permission to edit content

soda736 committed:
Adding in node check

soda736 committed:
PHPCS

soda736 committed:
PHPCS

soda736 committed:
Merge pull request #398 from dof-dss/CORP-472-origins

Corp 472 origins

soda736 committed:
Fixing depreciated code

soda736 committed:
Fixing translation block styling

soda736 committed:
Updating origins translation languages

soda736 committed:
Targetting menu with css

soda736 committed:
Merge pull request #399 from dof-dss/D8UN-3009-origins

D8 un 3009 origins

ooneill547 committed:
Add mp3 mimetype

ooneill547 committed:
Merge pull request #400 from dof-dss/CORP-613

CORP-613 ~ Add mp3 mimetype

omahm commented on origins_workflow/origins_workflow.module:
```diff
@@ -680,62 +681,72 @@ function origins_workflow_moderation_sidebar_alter(array &$build, EntityInterfac
+            // Add an 'Archive' button to the moderation sidebar, as
+            // it fails to do this in this case.
+            $build['actions']['primary']['archive'] = [];
+            $build['actions']['primary']['archive']['#title'] = Markup::create(t('Archive'));
```
Is the Markup call necessary?

neilblair commented on origins_workflow/origins_workflow.module:
```diff
@@ -680,62 +681,72 @@ function origins_workflow_moderation_sidebar_alter(array &$build, EntityInterfac
-        ];
+      $bundle = $entity->bundle();
+      // Check if a user has permission to edit the node content before giving them Archive and Draft of Published permission.
+      if ($bundle instanceof NodeType) {
```
Not sure this if block will ever run because `$entity->bundle()` returns a string (e.g. "Article") and so `if ($bundle instanceof NodeType)` will always be false.  So this code will never output a Draft of Published button.  I think this needs changed to:

```
      if ($entity instanceof Node) {
        $bundle = $entity->bundle();
```

neilblair replied:
I'll raise a separate PR to apply fixes to the dev branch for this

neilblair committed:
Update origins_workflow.module

- Set redirect destination to be internal node path instead of using the alias because if the change of state is archive, the alias redirect will not work leading to a 404
- Instead of checking whether use has permission to edit $bundle content (not working - maybe permission text is not right), check if the user has update access to the node
- Ensure the archive link triggers a confirmation to prevent accidentally archiving a node

neilblair committed:
Update origins_workflow.module

code stds

neilblair committed:
Update origins_workflow.module

code stds

neilblair committed:
Merge pull request #402 from dof-dss/D80-35

Update origins_workflow.module

neilblair committed:
Merge branch '10.x' into 10.x-dev
